### ISSUE TITLE
Suggest a different approach to `setupRouter` due to infinite recursion with accessing router service in EmberRouter.init

### ISSUE THREAD

**Issue by snewcomer** [ember-app-scheduler/ember-app-scheduler#985]
## Problem
Accessing `this.router` in `EmberRouter.init` will infinitely recurse due to the issue below.

https://github.com/emberjs/ember.js/issues/17791

```
export default class Router extends EmberRouter {
  location = config.locationType;
  rootURL = config.rootURL;

  constructor() {
    super(...arguments);

    setupRouter(this.router);
  },

  destroy() {
    reset();

    super.destroy(...arguments);
  }
}
```

I'm looking to fix this on Friday.  So perhaps this doesn't necessarily need to be fixed.  Thought I would create an issue for a paper trail.

**Comment by SergeAstapov** [comment]
@snewcomer @scalvert should we recommend to use instance initializer, similar to what `ember-router-scroll` addon is doing in https://github.com/DockYard/ember-router-scroll/blob/master/addon/instance-initializers/ember-router-scroll.js?

**Comment by snewcomer** [comment]
https://github.com/emberjs/ember.js/pull/19405

Luckily we fixed the infinite recursion in ember.js!  I'm not sure what we should do at this point though.  We could certainly refactor it but a fix is *coming.

**Comment by SergeAstapov** [comment]
@snewcomer right, it's fixed in Ember.js v3.26.0 https://github.com/emberjs/ember.js/blob/master/CHANGELOG.md#v3260-march-22-2021

I hit this in 3.24 so looks like this is not an issue anymore.
